addon/utils/bind-data-test-attributes.js

```diff
@@ -0,0 +1,13 @@
+const TEST_SELECTOR_PREFIX = /data-test-.*/;
+
+export default function bindDataTestAttributes(component) {
+  let attributeBindings = component.getWithDefault('attributeBindings', []);
+
+  for (let attr in component) {
+    if (TEST_SELECTOR_PREFIX.test(attr)) {
+      attributeBindings.push(attr);
+    }
+  }
+
+  component.set('attributeBindings', attributeBindings);
```

Comment:
that is true, but I'm not sure if `attributeBindings` being a CP is supported by Ember at all. otherwise there would be no need for us to apply the `bindDataTestAttributes()` function *before* the super constructor.